src/autosuggest/internal.tsx

```diff
@@ -86,7 +86,8 @@ const InternalAutosuggest = React.forwardRef((props: InternalAutosuggestProps, r
     warnOnce('Autosuggest', '`onLoadItems` must be provided for `recoveryText` to be displayed.');
   }
 
-  if (!enteredTextLabel) {
+  const enteredTextLabelI18nTestValue = i18n('enteredTextLabel', undefined, format => format({ value: '' }));
```

Comment:
Technically this is a fix, but it is not stable, because it does not give a solid coupling between the actual label rendering and this warning. The logic may get out of sync and there will be regressions.

Thanks for the contribution, I will take it from there and do additional refactoring

Reply:
For posterity: https://github.com/cloudscape-design/components/pull/1970